### DISCUSSION TITLE
Scrolling issue on mobile browsers

### DISCUSSION THREAD

**Discussion by zpydee** [tailwindlabs/tailwindcss#12226] [Help]
I've been pulling my hair out for 2 days so I'm turning to the community for help.

I've built a landing page in nextjs with tailwind and I'm encountering an issue where the page (sometimes) doesn't want to scroll I'm chrome and safari. Weirdly I can't get the problem to occur in Firefox.

When it does occur, I can scroll on the page edges, so I'm wondering if some of my tailwind code in my nested components is causing the problem .

To be honest though, I'm not even sure it's a tailwind problem. I'm certainly no CSS whizz though so maybe I'm missing something 

Has anyone run into anything similar?

**Comment by alex-krasikau** [comment]
Hello @zpydee,

There are a couple of scenarios in which this issue may occur. Here's what you can do:
1. First, examine all the elements that use `overflow: hidden` or the `hidden` class, and ensure that this behavior is intentional. In general, applying `overflow: hidden` to the body element can remove the scroll feature from the page.
2. Verify that you are not using any libraries that might be preventing scrolling. Sometimes, libraries used for rendering modal windows or popups, such as [Radix Dialog](https://www.radix-ui.com/primitives/docs/components/dialog), can block scrolling when you open a dialog.
3. Review your codebase for any instances of scroll events that might be interfering with scrolling.

If none of these steps resolve the issue, you can try a debugging approach. I typically start by systematically commenting out large sections of the React tree to identify the specific area causing the problem. Here's a method similar to binary search:
- Navigate to `app/layout.tsx` and comment out the `{children}` element. You can add static content directly to the layout to make it exceed the screen size. If scrolling works at this point, it suggests that the issue is related to the page's content.
- Next, proceed down to the page and start commenting out half of the JSX. Continue this process until you pinpoint the exact lines of code that are causing the scrolling problem.

Otherwise, we may need to have some kind of reproduction of this issue to help with it further.

**Reply by zpydee** [reply]
Thanks @alex-krasikau. Let me try this approach and see if I make any progress. If not, i'll create a repro and share it.

**Reply by MahdiKabir** [reply]
Were you able to figure out what was causing this? Running into exact the same issue.
